### PR TITLE
SES-697 - Add loading state when exporting logs

### PR DESCRIPTION
**Pull request by alansley**
### Contributor checklist
- [X] I have tested my contribution on these devices:
 * Virtual device Pixel 3a, Android 9 / API 28
 * Virtual device Pixel 3a, Android 14 / API 34
- [X] My contribution is fully baked and ready to be merged as is
- [X] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description

Adds a progress bar to the Help | Report a Bug | [Share Logs] facility. Changed the [Export Logs] button text to [Cancel] while in progress. Losing focus on the job fragment removes the progress bar & puts the "Export Logs" text back, as does the job being cancelled or the job completing.

Also caught an AvatarHelper exception and printed less verbose msgs as it hits and prints a full stack trace all the time!

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt [hunk 1]
```diff
@@ -35,20 +43,48 @@ import java.util.concurrent.TimeUnit
...
+            // Change export logs button text
+            val exportLogsButton = this.activity?.findViewById(R.id.export_logs_button) as TextView?
+            if (exportLogsButton == null) { Log.w("Loki", "Could not find export logs button view.") }
+            exportLogsButton?.text = if (exportJobRunning) getString(R.string.cancel) else getString(R.string.activity_help_settings__export_logs)
+
+            // Show progress bar
+            val exportProgressBar = this.activity?.findViewById(R.id.export_progress_bar) as ProgressBar?
+            // Note: I'm not using `isVisible` because those options are VISIBLE or GONE and I want
+            // to keep taking the space as INVISIBLE to avoid any layout change on show/hide.
+            exportProgressBar?.visibility = if (exportJobRunning ) VISIBLE else INVISIBLE
```
[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt:73]: Ah! Nice - now used.

File: app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt [hunk 2]
```diff
@@ -95,13 +134,36 @@ class ShareLogsDialog : DialogFragment() {
...
                     startActivity(Intent.createChooser(shareIntent, getString(R.string.share)))
                 }
-
-                dismiss()
             } catch (e: Exception) {
                 withContext(Main) {
                     Log.e("Loki", "Error saving logs", e)
                     Toast.makeText(context,"Error saving logs", Toast.LENGTH_LONG).show()
                 }
+                dismiss()
```
[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt:142]: Good point :)

File: libsession/src/main/java/org/session/libsession/avatars/AvatarHelper.java [hunk 3]
```diff
@@ -22,9 +24,15 @@ public class AvatarHelper {
...
   public static InputStream getInputStreamFor(@NonNull Context context, @NonNull Address address)
-      throws IOException
   {
-    return new FileInputStream(getAvatarFile(context, address));
+    try {
+      return new FileInputStream(getAvatarFile(context, address));
+    }
+    catch (FileNotFoundException fnfe) {
+      // Catching this to prevent excessively verbose error output in LogCat
+      Log.e("Loki-Avatar", "Uploading avatar failed for ID (print this stack trace for the gory details): " + address);
```
[the remark this answers is not in the document]
Comment (alansley) [reply libsession/src/main/java/org/session/libsession/avatars/AvatarHelper.java:33]: Done

File: app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt [hunk 4]
```diff
@@ -35,20 +43,48 @@ import java.util.concurrent.TimeUnit
...
     }
 
-    private fun shareLogs() {
+    // If the share logs dialog loses focus the job gets cancelled so we'll update the UI state
+    override fun onPause() {
+        super.onPause()
+        updateExportButtonAndProgressBarUI(false)
+    }
+
+    private fun updateExportButtonAndProgressBarUI(exportJobRunning: Boolean) {
```
[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt:62]: Done

**Comment by alansley** [comment]
All feedback addressed & should be ready to go :)

**Comment by burtonemily** [comment]
Tested and working :)

**Pull request merged**